File: cache/src/main/java/com/osrsd/cache/def/ParamDefinition.java
package com.osrsd.cache.def;

import lombok.Data;

@Data
public final class ParamDefinition implements Definition {

    private int id;
    private char type;
    private boolean isMembers;
    private int defaultInt;
    private String defaultString;

    public ParamDefinition(int id) {
        this.id = id;
        isMembers = true;
    }

}

File: app/src/main/kotlin/com/osedit/spring/domain/ParamDefinition.kt
package com.osedit.spring.domain

import com.osedit.cache.Config
import com.osedit.cache.Indices
import com.osedit.cache.provider.ParamProvider
import java.nio.ByteBuffer
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class ParamDefinition(@Id private val id: Int? = 0) : Definition {
    var type: Char? = null
    var isMembers: Boolean = true
    var defaultInt: Int = 0
    var defaultString: String? = null

    override fun path(): String {
        return "/params"
    }

    override fun config(): Config {
        return Config.PARAMS
    }

    override fun indices(): Indices {
        return Indices.CONFIG
    }

    override fun decode(byteBuffer: ByteBuffer): Definition {
        return ParamProvider.decode(byteBuffer, this)
    }
}